src/command-helpers/scaffold.js

```diff
@@ -59,8 +79,54 @@ const writeScaffold = async (scaffold, directory, spinner) => {
   await writeScaffoldDirectory(scaffold, directory, spinner)
 }
 
+const writeABI = async (abi, contractName, abiPath) => {
+  let data = prettier.format(JSON.stringify(abi.data), {
+    parser: 'json',
+  })
+  let filePath = abiPath ? abiPath : `./abis/${contractName}.json`
+
+  await fs.writeFile(filePath, data, { encoding: 'utf-8' })
+}
+
+const writeSchema = async (abi, protocol, schemaPath, entities) => {
+  const events = protocol.hasEvents()
+    ? abiEvents(abi).filter(event => entities.indexOf(event.get('name')) === -1).toJS()
```

Comment:
The reason you do a `.toJS()` here is something to due with the `prettier` API?

Reply:
No, it's because `abi.data` is an immutable.js type and if I don't turn it into a JS type the `generateEventType` function crashes:

![image](https://user-images.githubusercontent.com/22292075/168100033-693623d0-2b7a-4f56-a1b4-70814f475565.png)